librz/core/core.c

```diff
@@ -2446,3 +2446,31 @@ RZ_API RzCmdStatus rz_core_core_plugins_print(RzCore *core, RzCmdStateOutput *st
 	rz_cmd_state_output_array_end(state);
 	return RZ_CMD_STATUS_OK;
 }
+
+/**
+ * \brief Filters the given string based on the provided filter.
+ *
+ * \param str RZ_NONNULL The string to be filtered.
+ * \param filter The filter string to be used for filtering the str.
+ * \return RZ_OWN char* The filtered string. The caller is responsible for freeing this string.
+ */
+RZ_API RZ_OWN char *rz_core_filter_string_output(RZ_NONNULL const char *str, const char *filter) {
```

Comment:
Rename into `rz_str_filter_apply()` and move into `librz/util/str.c`
Would be nice also if you add Doxygen description to `rz_str_filter_zeroline()` so there is no confusion.